### ISSUE TITLE
Potential randomness issues

### ISSUE THREAD

**Issue by adamunchained** [soonlabs/soonaverse-dao#630]
Container for list of issues that might potentially need to be looked when it comes to randomness (when buying NFT). Please post your finding so we can improve.

Source: https://discord.com/channels/892481940953915433/908575339951833129/964547703914758155

**Comment by csxiong** [comment]
I also noticed this during my Mint of NF3 in the last few days, after the number of available soonanaut was below 1000, I Minted about 30 of them and basically over 90% of the numbers were less than 1000. 

We can find out if this randomness issue does exist by checking all the records of NF3's mint.

**Comment by adamunchained** [comment]
Thanks @csxiong - I’ll be investigating and provide an update later today

**Comment by skay8** [comment]
Interesting. 
i noticed the same and it explains all my thoughts.

**Comment by baylander** [comment]
![grafik](https://user-images.githubusercontent.com/29330260/163627410-57e2d91b-c785-4838-a27b-8600bc640281.png)
Checking the numbers, it seems that "nightfall" and "ascension" nft have been randomly picked after one hour, about double of their probability.

Est.Probability after 1h and  "2339" total mints.
0,5		1169,5	(Palm Springs)
0,3		701,7	(Floating..)
0,1		233,9	(Tranquility)
0,06		140,34 	(Ascension)
0,04		93,56 	(Nightfall)

Data Source: Extract from datastudio ("Soonaverse information....")
- Tried to get rid of double entries (nft swaps by owners) by deleting all value !=500Mi

**Comment by adamunchained** [comment]
This has been fixed in production. 

NFT is picked randomly at the order stage. It uses position of a NFT within collection as an input to increase randomness. The position was not correctly set within NFT records and it significantly reduced (not completely) randomness. Typically creators would randomize their collection (properties/stats) as part of their upload but unfortunately this not has happen today and it made this issue apparent. 

I've now deployed fix in production to make sure position is correctly set and also corrected position stored on existing NFTs.
